graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/acl/service/JahiaAclServiceImpl.java

```diff
@@ -121,6 +122,25 @@ public boolean hasInheritedPermission(JCRNodeWrapper jcrNode, String principalKe
         return aclEntries.stream().anyMatch(ace -> roleName.equals(ace.getRoleName()) && ace.isGrantType() && ace.isInherited());
     }
 
+    public boolean hasInheritedUserRole(JCRNodeWrapper node, JahiaUser user, String roleName) throws RepositoryException {
+        List<JahiaAclEntry> aclEntries = this.getAclEntries(node);
+        JCRSiteNode site = node.getResolveSite();
+        return aclEntries
+                .stream()
+                .anyMatch(ace -> {
+                    if (!ace.isGrantType()) return false;
+                    if (ace.getRoleName().equalsIgnoreCase(roleName)) {
+                        if (ace.getPrincipalKey().startsWith("u:") && ace.getPrincipalKey().substring(2).equalsIgnoreCase(user.getUsername())) {
```

Comment:
Could be replaced with `PrincipalType.isUser(ace.getPrinicipalKey())`, and also consider adding an ace function `ace.isUserPrincipal()` using PrincipalType.isUser